src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -749,38 +756,62 @@ private bool IsPublicAPI(ISymbol symbol)
                     return false;
                 }
 
-                return IsPublicApiCore(symbol);
+                return IsTrackedApiCore(symbol);
             }
 
-            private bool IsPublicApiCore(ISymbol symbol)
+            private bool IsTrackedApiCore(ISymbol symbol)
             {
-                return symbol.DeclaredAccessibility switch
-                {
-                    Accessibility.Public => symbol.ContainingType == null || IsPublicApiCore(symbol.ContainingType),
-                    Accessibility.Protected
-                    or Accessibility.ProtectedOrInternal => symbol.ContainingType != null
-                        && IsPublicApiCore(symbol.ContainingType)
-                        && CanTypeBeExtendedPublicly(symbol.ContainingType),// Protected symbols must have parent types (that is, top-level protected
-                                                                            // symbols are not allowed.
-                    _ => false,
-                };
+                var resultantVisibility = symbol.GetResultantVisibility();
```

Comment:
The original logic was very complicated to adapt to the `internal` world. Really what we care about is twofold:

1. Is the API effectively target accessibility?
2. If the API is protected (or protected internal for target accessibility == public), can that protected api actually be observed outside the current assembly?